### PR TITLE
Reconnect to Nats in publisher-proxy

### PR DESCRIPTION
**Pull request by radufa**
**Description**

Changes proposed in this pull request:

- Reconnect if the Nats connection was closed during pod migrations

**Related issue(s)**

### CODE REVIEW

File: components/event-publisher-proxy/pkg/nats/connect.go [hunk 1]
```diff
@@ -7,14 +7,40 @@ import (
...
+}
+
+type BackendConnection struct {
+	connectionData connectionData
+	Connection     *nats.Conn
+}
+
+// NewBackendConnection returns a new NewNatsConnection instance with the given NATS connection data.
+func NewBackendConnection(url string, retry bool, reconnects int, wait time.Duration) *BackendConnection {
+	bc := &BackendConnection{}
```
Comment (themue) [inline components/event-publisher-proxy/pkg/nats/connect.go:24]:
```suggestion
	return &BackendConnection{
	    connectionData: connectionData{
	        url: url,
	        retry: retry,
	        reconnects: reconnects:,
	        wait: wait,
	    },
	}
```

File: components/event-publisher-proxy/pkg/sender/nats.go [hunk 2]
```diff
@@ -7,7 +7,7 @@ import (
 	cenats "github.com/cloudevents/sdk-go/protocol/nats/v2"
 	cev2 "github.com/cloudevents/sdk-go/v2"
 	cev2event "github.com/cloudevents/sdk-go/v2/event"
-	"github.com/nats-io/nats.go"
+	pkgnats "github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/nats"
```
Comment (themue) [inline components/event-publisher-proxy/pkg/sender/nats.go:10]: Possibly a different name than `pkgnats` as they are all somehow `pkg`. Maybe `epnats` like `cenats` above.

**Review by themue — COMMENTED** [review]
Only two minor comments.

File: components/event-publisher-proxy/pkg/sender/nats.go (hunk 2 above)
Comment (radufa) [reply components/event-publisher-proxy/pkg/sender/nats.go:10]: "pkgnats" was defined and used in the past, see:
https://github.com/kyma-project/kyma/blob/main/components/event-publisher-proxy/cmd/event-publisher-proxy/nats/nats.go#L13
https://github.com/kyma-project/kyma/blob/main/components/event-publisher-proxy/pkg/handler/nats/handler_test.go#L30
https://github.com/kyma-project/kyma/blob/main/components/event-publisher-proxy/pkg/sender/nats_test.go#L15

Comment (themue) [reply components/event-publisher-proxy/pkg/sender/nats.go:10]: Okay, so out of scope. 👌🏻

**Review by themue — APPROVED** [review]

**Pull request merged**